subsys/bluetooth/audio/ascs.c

```diff
@@ -236,37 +241,145 @@ static int ascs_disconnect_stream(struct bt_bap_stream *stream)
 				 K_MSEC(CONFIG_BT_ASCS_ISO_DISCONNECT_DELAY));
 }
 
-void ascs_ep_set_state(struct bt_bap_ep *ep, uint8_t state)
+static void ase_set_state_idle(struct bt_ascs_ase *ase)
 {
-	struct bt_bap_stream *stream;
-	bool state_changed;
-	uint8_t old_state;
+	struct bt_bap_stream *stream = ase->ep.stream;
 
-	if (!ep) {
-		return;
+	__ASSERT_NO_MSG(stream != NULL);
+
+	ase->ep.receiver_ready = false;
+
+	ase_status_changed(ase, BT_BAP_EP_STATE_IDLE);
+
+	bt_bap_stream_reset(stream);
+
+	if (stream->ops != NULL && stream->ops->released != NULL) {
+		stream->ops->released(stream);
 	}
 
-	/* TODO: Verify state changes */
+	ase_free(ase);
+}
+
+static void ase_set_state_codec_configured(struct bt_ascs_ase *ase)
+{
+	struct bt_bap_stream *stream = ase->ep.stream;
 
-	old_state = ep->status.state;
-	ep->status.state = state;
-	state_changed = old_state != state;
+	__ASSERT_NO_MSG(stream != NULL);
 
-	LOG_DBG("ep %p id 0x%02x %s -> %s", ep, ep->status.id, bt_bap_ep_state_str(old_state),
-		bt_bap_ep_state_str(state));
+	ase->ep.receiver_ready = false;
 
-	/* Notify clients*/
-	ase_status_changed(ep, old_state, state);
+	ase_status_changed(ase, BT_BAP_EP_STATE_CODEC_CONFIGURED);
 
-	if (ep->stream == NULL) {
-		return;
+	if (stream->ops != NULL && stream->ops->configured != NULL) {
+		stream->ops->configured(stream, &ase->ep.qos_pref);
 	}
+}
```

Comment:
General comment: Consider storing `stream->ops` in a local variable to reduce number of dereferences (not for performance, but rather just code readability).

Applies multiple places